pkg/plugins/common/kustomize/v2/scaffolds/init.go

```diff
@@ -69,6 +69,9 @@ func (s *initScaffolder) Scaffold() error {
 		&rbac.AuthProxyService{},
 		&rbac.AuthProxyClientRole{},
 		&rbac.RoleBinding{},
+		// We need to create a Role because if the project
+		// has not CRD define the controller-gen will not generate this file
+		&rbac.Role{},
```

Comment:
It is not under the scope of this PR
The fix done in https://github.com/kubernetes-sigs/kubebuilder/pull/3626 is here otherwise that fails with the grafana sample